raft/multinode.go

```diff
@@ -143,24 +143,23 @@ func (g *groupState) commitReady(rd Ready) {
 	if !IsEmptyHardState(rd.HardState) {
 		g.prevHardSt = rd.HardState
 	}
-	if !IsEmptySnap(rd.Snapshot) {
-		g.prevSnapi = rd.Snapshot.Metadata.Index
-		g.raft.raftLog.stableSnapTo(g.prevSnapi)
+	if g.prevHardSt.Commit != 0 {
+		// In most cases, prevHardSt and rd.HardState will be the same because
```

Comment:
IIRC, Only the prevHardSt here provides the most accurate information to applyTo. Either CommittedEntries or HardState may miss the initial commitindex state in some cases.